### ISSUE TITLE
get_sonar_distance issues

### ISSUE THREAD

**Issue by tonySRP** [Krakenus/microbit-cutebot-micropython#2]
Hi,

When i try to use the servo and sonar it comes up with a name error.  Not sure if you have any idea why.  WOuld it be possible for you to upload a couple of examples of code that show these functions work.

Thanks a lot

Tony

**Comment by Krakenus** [comment]
@tonySRP
Hello, thank you for your feedback. I will check it. 

I know it's hard to read errors from microbit display but can you please post here the line where you got the error?

**Comment by Krakenus** [comment]
@tonySRP 
Hi again,
So far I checked the code and I don't think there is a place where the NameError could show up. 
Do you import the cutebot module before you use it's functions?

It should be like this:
```
import cutebot

cutebot.set_servo_1_angle(42)
cutebot.get_sonar_distance()
```

Unfortunately I won't have enough free time to test it with hardware during this week, but if you can send me some of your code I think I can help.

**Comment by tonySRP** [comment]
Hi Milan,

I will replicate the issue and take a video. Thanks for replying. What you have done here is brilliant and I really appreciate what you have made.

Thanks

Tony Cornes
________________________________

**Comment by Krakenus** [comment]
Hello @tonySRP ,
I just pushed some examples for sonar and servo so you can try them on your cutebot.

https://github.com/Krakenus/microbit-cutebot-micropython/blob/master/examples/sonar_distance.py
https://github.com/Krakenus/microbit-cutebot-micropython/blob/master/examples/servo_control.py
I tried both of them without any error. 

Please note that I also added some new constants to `cutebot.py` so make sure you are using the latest version.

Anything new on your side? If you still get errors I'm afraid I will need to see your code.

**Comment by tonySRP** [comment]
Thanks for this. Sorry haven’t had chance to test it this week. I will check next week. Thanks again.

Tony Cornes
________________________________

**Comment by Krakenus** [comment]
@tonySRP Hello, any luck with examples?

**Comment by spkuehl** [comment]
@Krakenus I am trying to use sonar distancing on micro:bit cutebot but running your code returns -2 for me in the loop. Can't figure out why.

**Comment by spkuehl** [comment]
If I change the pins in `get_sonar_distance` I can get it to return -1, but nothing else.

**Comment by Krakenus** [comment]
@spkuehl 
Hello,
thanks for your feedback. I'm afraid my sonar module is broken so I have to get a new one before I'll be able to test it again. 

However negative number returned from this function which is called from `get_sonar_distance` so it looks like that your sonar module doesn't make any pulse to catch during the distance measuring.
http://docs.micropython.org/en/latest/library/machine.html?highlight=time_pulse_us#machine.time_pulse_us

Does your sonar module work when you try this example?
https://www.elecfreaks.com/learn-en/microbitKit/smart_cutebot/cutebot_case09.html

**Comment by spkuehl** [comment]
Yes the sonar does work with the elecfreaks sample cases. I have a brand new microbit and cutebot, maybe something changed on the hardware recently.

@Krakenus

**Comment by Krakenus** [comment]
@spkuehl I don't think they would change the connection just like that.

Are you able to test the module without cutebot? Using Arduino for example?

**Comment by Krakenus** [comment]
I think I found the issue in the code.

I should receive new HC-SR04 sonar module tomorow, so I will test it the comming days.

**Comment by Krakenus** [comment]
@spkuehl
Hello, I did some testing and it looks like the code is ok after all but for me the problem is in powering the sonar module. The operational voltage of HC-SR04 is 5V. The cutebot's batteries should provide voltage about 4.5V which should be ok but I measured only 3.3V even with brand new batteries.

I tried equivalent code with arduino which is using 5V logic and it worked.
